src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/PackageValidatorFactory.cs

```diff
@@ -28,14 +28,18 @@ public PackageValidatorFactory(ILoggerFactory loggerFactory)
             StorageFactory auditingStorageFactory,
             IEnumerable<EndpointFactory.Input> endpointInputs,
             Func<HttpMessageHandler> messageHandlerFactory,
+            bool expectsSignature = false,
             bool verbose = false)
         {
             var validatorFactory = new ValidatorFactoryFactory(_loggerFactory).Create(galleryUrl, indexUrl);
-
             var endpointFactory = new EndpointFactory(validatorFactory, messageHandlerFactory, _loggerFactory);
-            var endpoints = endpointInputs.Select(e => endpointFactory.Create(e));
 
-            return new PackageValidator(endpoints, auditingStorageFactory, _loggerFactory.CreateLogger<PackageValidator>());
+            var validators = new List<IAggregateValidator>();
+
+            validators.AddRange(endpointInputs.Select(e => endpointFactory.Create(e)));
+            validators.Add(new CatalogAggregateValidator(validatorFactory, expectsSignature));
```

Comment:
You should have tests for `CatalogAggregateValidator`, but wouldn't it be simpler to do away with `CatalogAggregateValidator` and conditionally add the validator here?

Reply:
It wouldn't be simpler, no. `IAggregateValidator` and `IValidator` return different values. The thing that runs validators expects them to be wrapped by an `IAggregateValidator` so that an `AggregateValidationResult` is returned.